modules/aruco/src/aruco.cpp

```diff
@@ -1542,22 +1542,22 @@ void _drawPlanarBoardImpl(Board *_board, Size outSize, OutputArray _img, int mar
             // remove negativity
             p1.x = p0.x - minX;
             p1.y = p0.y - minY;
-            pf.x = p1.x * float(markerZone.cols - 1) / sizeX;
-            pf.y = float(markerZone.rows - 1) - p1.y * float(markerZone.rows - 1) / sizeY;
+            pf.x = p1.x * float(markerZone.cols) / sizeX;
+            pf.y = float(markerZone.rows - 1) - p1.y * float(markerZone.rows) / sizeY;
```

Comment:
and this should be rewritten as `markerZone.rows - p1.y * float(markerZone.rows) / sizeY - 1` for readability